crux-core/src/crux/api/document/CruxId.java

```diff
@@ -0,0 +1,44 @@
+package crux.api.document;
+
+import clojure.lang.IPersistentMap;
+import clojure.lang.Keyword;
+import crux.api.exception.CruxIdException;
+
+import java.net.URI;
+import java.net.URL;
+import java.util.Objects;
+import java.util.UUID;
+
+public class CruxId {
```

Comment:
The `CruxId` class doesn't allow allow instances for this reason as we discussed. 

Atm, it is basically a place for static type assertions to live.

I think it could be worth continuing to have these runtime checks when the transaction objects are created, as potentially this could be done far in advance of them actually being run on a node.